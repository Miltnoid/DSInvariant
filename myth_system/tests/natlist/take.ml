type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let take : nat -> list -> list |>
  { 0 => ( [] => []
         | [1] => []
         | [0;1] => []
         | [1;0;1] => [] )
  | 1 => ( []    => []
         | [1]   => [1]
         | [0;1] => [0]
         | [1;0;1] => [1] )
  | 2 => ( []    => []
         | [1]   => [1]
         | [0;1] => [0;1]
         | [1;0;1] => [1;0] )
  } = ?
