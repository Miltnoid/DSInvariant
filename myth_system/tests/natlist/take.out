(* Generated Wed Dec 31 11:16:14 2014 *)
let take : nat -> list -> list =
  let rec f1 (n1:nat) : list -> list =
    fun (l1:list) ->
      match n1 with
        | O -> Nil
        | S n2 -> (match l1 with
                     | Nil -> Nil
                     | Cons (n3, l2) -> Cons (n3, f1 n2 l2))
  in
    f1
;;
